addons/sale_timesheet/models/account.py

```diff
@@ -21,22 +21,29 @@
 class AccountAnalyticLine(models.Model):
     _inherit = 'account.analytic.line'
 
-    def _default_sale_line_domain(self):
-        domain = super(AccountAnalyticLine, self)._default_sale_line_domain()
-        return expression.OR([domain, [('qty_delivered_method', '=', 'timesheet')]])
-
     timesheet_invoice_type = fields.Selection(TIMESHEET_INVOICE_TYPES, string="Billable Type",
             compute='_compute_timesheet_invoice_type', compute_sudo=True, store=True, readonly=True)
     commercial_partner_id = fields.Many2one('res.partner', compute="_compute_commercial_partner")
     timesheet_invoice_id = fields.Many2one('account.move', string="Invoice", readonly=True, copy=False, help="Invoice created from the timesheet")
     so_line = fields.Many2one(compute="_compute_so_line", store=True, readonly=False,
-        domain="[('is_service', '=', True), ('is_expense', '=', False), ('state', '=', 'sale'), ('order_partner_id', 'child_of', commercial_partner_id)]",
         help="Sales order item to which the time spent will be added in order to be invoiced to your customer. Remove the sales order item for the timesheet entry to be non-billable.")
     # we needed to store it only in order to be able to groupby in the portal
     order_id = fields.Many2one(related='so_line.order_id', store=True, readonly=True, index=True)
     is_so_line_edited = fields.Boolean("Is Sales Order Item Manually Edited")
     allow_billable = fields.Boolean(related="project_id.allow_billable")
 
+    def _default_sale_line_domain(self):
+        return expression.OR([[
+            ('is_service', '=', True),
+            ('is_expense', '=', False),
+            ('state', 'in', ['sale', 'done']),
```

Comment:
```suggestion
            ('state', '=', 'sale'),
```

`done` state doesn't exist anymore